public/styles.css

```diff
@@ -0,0 +1,832 @@
+  padding: 1rem;
+}
+
+.small-text {
+  font-size: 0.7rem;
+}
+
+.text-left {
+  text-align: left;
+}
+.text-center {
+  text-align: center;
+}
+
+.text-right {
+  text-align: right;
+}
+
+.bold-text {
+  font-weight: 700;
+}
+
+.light-bold-text {
+  font-weight: 600;
+}
+
+.light-text {
+  font-weight: 300;
+}
+
+.grey-text {
+  font-weight: 100;
+}
+
+.darkmode-icon {
+  transform: rotate(15deg);
+}
+
+.icon-github {
+  width: 25px;
+  height: auto;
+  margin: 0px 0.5rem;
+  opacity: 0.9;
+}
+
+.icon-ham {
+  cursor: pointer;
+  width: 1.25rem;
+  height: auto;
+  padding-right: 5px;
+  padding-top: 1px;
+}
+
+.hide {
+  display: none;
+}
+
+.search-box {
+  padding-left: 1rem;
+  border: 0;
+  border-radius: 5px;
+  box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 3px 0px,
+    rgba(0, 0, 0, 0.06) 0px 1px 2px 0px;
+}
+
+.homepage {
+  background-color: white;
+  z-index: 200;
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  width: 100%;
+  height: 100%;
+  overflow: hidden;
+}
+
+.homepage-text {
+  color: #023e8a;
+  margin: 1rem 0rem;
+  text-align: center;
+}
+
+.hide-homepage {
+  display: none;
+}
+
+@media only screen and (max-width: 851px) {
+  .homepage-content {
+    margin-top: -1rem;
+    width: 100vw;
+    height: 100vh;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+  }
+
+  .homepage-text {
+    width: 80%;
+    font-size: 60%;
+    overflow: hidden;
+  }
+
+  .homepage-text h1 {
+    font-weight: 400;
+  }
+
+  .homepage-logo {
+    width: 4rem;
+  }
+  .homepage-logo-text {
+    width: 8rem;
+  }
+  .button-homepage-primary,
+  .button-homepage-secondary {
+    font-size: 0.7rem;
+    margin: 1rem;
+    padding: 0.5rem;
+  }
+}
+
+@media only screen and (min-width: 600px) {
+  .grid-view {
+    display: grid;
+    grid-template-areas: "side-bar content";
+    grid-template-columns: 1fr 3.5fr;
+    grid-gap: 1rem;
+  }
+
+  .header {
+    margin-bottom: 10rem;
+  }
+
+  .side-bar {
+    margin-top: -2rem;
+    grid-area: "side-bar";
+    position: fixed;
+    font-size: 1.1rem;
+    height: 100vh;
+    overflow-y: auto;
+    scrollbar-width: thin;
+  }
+  .content {
+    margin-top: 6.5rem;
+    grid-area: "content";
+    padding-left: 13rem;
+  }
+  .navBar {
+    display: none;
+  }
+}
+
+@media only screen and (min-width: 851px) {
+  /* .grid-view,
+  .header-content,
+  .footer-content {
+    margin: 0rem 15rem 0rem 10rem;
+  } */
+  .logo {
+    margin-left: 1.7rem;
+    width: 35px;
+  }
+
+  .icon-github {
+    width: 30px;
+    opacity: 0.8;
+    margin-top: 0.5rem;
+  }
+  .hide {
+    display: block;
+  }
+
+  .logo-text {
+    padding: 0rem;
+    margin-top: 0.3rem;
+    margin-left: 1rem;
+    width: 4.8rem;
+    height: auto;
+  }
+
+  .grid-header {
+    display: grid;
+    grid-template-areas: "column-header column-header column-header";
+    grid-template-columns: 3.5fr 2fr auto;
+  }
+  .column-header {
```

Comment:
general comment and not gonna spam here, just go through all the code and add spacing wherever needed. But take care of extra spaces as well.